syft/pkg/cataloger/javascript/cataloger.go

```diff
@@ -23,5 +31,47 @@ func NewJavascriptLockCataloger() *common.GenericCataloger {
 		"**/yarn.lock":         parseYarnLock,
 	}
 
-	return common.NewGenericCataloger(nil, globParsers, "javascript-lock-cataloger")
+	return common.NewPostProcessingGenericCataloger(nil, globParsers, "javascript-lock-cataloger", addLicenses)
+}
+
+func addLicenses(resolver source.FileResolver, location source.Location, p *pkg.Package) {
+	dir := path.Dir(location.RealPath)
+	pkgPath := []string{dir, "node_modules"}
+	pkgPath = append(pkgPath, strings.Split(p.Name, "/")...)
+	pkgPath = append(pkgPath, "package.json")
+	pkgFile := path.Join(pkgPath...)
+	locations, err := resolver.FilesByPath(pkgFile)
+	if err != nil || len(locations) == 0 {
+		log.Debugf("no package.json found at: %s", pkgFile)
```

Comment:
this could be a little noisy, is it necessary to log that there is no package.json?

Also, should we put out a warning when there is an error? I feel that these cases should be handled separately.